### PR TITLE
feat: extract Slack + GitHub provisioning into services + connect-slack/connect-github routes (UAP-1.1, UAP-1.2)

### PR DESCRIPTION
**Pull request by dodizzle**
## Summary

**UAP-1.1 — Slack provisioning extraction**
- Extracted the Slack app-manifest-creation/OAuth/app-token orchestration out of `admin/src/admin-ui.ts`'s inline `/admin/provision/*` handlers into a new `SlackProvisioningService` (`admin/src/slack-provisioning-service.ts`)
- Added three new per-agent routes that delegate to the same service: `POST /admin/agents/:id/connect-slack`, `GET /admin/agents/:id/connect-slack/callback`, `POST /admin/agents/:id/connect-slack/app-token`
- The existing `/admin/provision/start` (Slack branch), `/admin/provision/complete`, and `/admin/provision/xapp-token` handlers now delegate to the same extracted service — pure extraction, no behavior change for the not-yet-retired wizard

**UAP-1.2 — GitHub provisioning extraction**
- Extracted the GitHub App manifest/install flow and the PAT-only path out of `admin/src/admin-ui.ts`'s inline `/admin/provision/start`, `/admin/provision/github-app/complete`, `/admin/provision/github-app/installed` handlers into a new `GithubProvisioningService` (`admin/src/github-provisioning-service.ts`), built on the existing `github-app-provisioning-client.ts`
- Added three new per-agent routes that delegate to the same service: `POST /admin/agents/:id/connect-github`, `GET /admin/agents/:id/connect-github/callback`, `GET /admin/agents/:id/connect-github/installed`
- The existing `/admin/provision/start`'s GitHub branches and `/admin/provision/github-app/complete` / `installed` handlers now delegate to the same extracted service — pure extraction, no behavior change for the not-yet-retired wizard

## Acceptance Criteria

**UAP-1.1**

| # | Criterion | Status |
|---|-----------|--------|
| 1 | New service function(s) exist with extracted logic, following architecture_layering | MET |
| 2 | Three new connect-slack routes exist and reach the same successful outcome as the legacy Slack path | MET |
| 3 | Existing `/admin/provision/start`, `/complete`, `/xapp-token` behave identically, now delegating to the same service | MET |
| 4 | Integration test for the service (extends `slack-provisioning.integration.test.ts`'s fixture pattern) + smoke test for the three new routes | MET |

**UAP-1.2**

| # | Criterion | Status |
|---|-----------|--------|
| 1 | New service function(s) wrap github-app-provisioning-client.ts and the PAT-storage path; admin-ui.ts route handlers only parse params and call the service | MET |
| 2 | Three new connect-github routes exist and produce the same outcomes as the legacy GitHub App/PAT routes, across all three modes (pat, app-auto, app-manual) | MET |
| 3 | Existing `/admin/provision/start`'s GitHub branches and `/admin/provision/github-app/complete` / `installed` continue to behave identically, now delegating to the same extracted service | MET |
| 4 | Smoke test for the three new routes covering all three modes (pat/app-auto/app-manual); `github-app-provisioning-client.integration.test.ts` untouched | MET |

## Test Plan
- `admin/src/slack-provisioning-service.integration.test.ts` — 13 tests against the recorded Slack-client fixture double, covering `startConnect`/`completeConnect`/`saveAppToken` happy paths and edge cases (missing/expired state, missing code, OAuth exchange failure, reinstall path, agent-not-found, invalid token prefixes)
- `admin/src/admin-ui.smoke.test.ts` — 7 smoke tests for the connect-slack routes, plus 18 new smoke tests for the three connect-github routes (pat/app-manual/app-auto modes, callback, installed) via in-process `app.request()`
- `github-provisioning-service.ts` at 100% line/function coverage
- Existing `/admin/provision/*` smoke/integration coverage untouched and still passing (no regression)
- `task typecheck`, `bunx biome lint`, and `task ci`'s admin-relevant suites all pass; remaining `task ci` failures are pre-existing on `main` (`task-store`/`metrics` packages, unrelated to this change)
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Review by dmcaulay — COMMENTED** [review]
Verdict: COMMENT — docs/agent.md's new github-provisioning-service.ts reference row documents createAppManifest/startConnect/saveAppCredentials, but the actual exports are startPatConnect/startAppManualConnect/startAppAutoConnect/completeConnect/completeInstalled/signProvisionState/readProvisionState. The extraction itself is behavior-preserving and well-tested.

File: docs/agent.md [hunk 1]
```diff
@@ -286,6 +286,8 @@ The constant `BAKED_MARKETPLACES_ROOT` and function `discoverBakedMarketplaces()
...
...b`, `GET /admin/agents/:id/connect-github/callback`, `GET /admin/agents/:id/connect-github/installed`). Exports `GITHUB_PROVISION_STATE_COOKIE`, `GITHUB_PROVISION_STATE_TTL_SECONDS`, and `GITHUB_ORG_PATTERN` constants (shared with admin-ui.ts routes); core methods: `createAppManifest(agentName, githubOrg, opts?)` — builds GitHub App manifest for the agent with configurable redirect/setup URIs; `signProvisionState(agentId, githubOrg)` — JWT signer for the provision-state cookie; `startConnect(agentId, githubOrg)` — validates agent existence, calls `createAppManifest()`, and returns a signed token + GitHub manifest URL; `completeConnect(rawStateCookie, code)` — verifies the provision-state token, exchanges the manifest code for app credentials (via `GithubAppProvisioningClient`), and returns outcome: `invalid_state`, `missing_code`, `exchange_failed`, or `success`; `saveAppCredentials(agentId, credentials)` — stores GH_TOKEN / GH_APP_ID / GH_APP_INSTALLATION_ID / GH_APP_PRIVATE_KEY and reconciles system crons. No HTTP mechanics (reading/writing cookies, redirects, HTML rendering) — those stay in the route handlers. Mirrors `SlackProvisioningService`'s DI/result-union shape exactly. |
```
Comment (dmcaulay) [inline docs/agent.md]: This row documents `createAppManifest(agentName, githubOrg, opts?)`, `startConnect(agentId, githubOrg)`, and `saveAppCredentials(agentId, credentials)` as GithubProvisioningService's core methods, but none of these exist. The actual exported methods (confirmed via `grep '^export'` on github-provisioning-service.ts) are `startPatConnect`, `startAppManualConnect`, `startAppAutoConnect`, `signProvisionState`, `readProvisionState`, `completeConnect`, and `completeInstalled`. Please update this row to describe the real API surface.

File: admin/src/slack-provisioning-service.ts [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+  /** Env var keys that must be stored as secrets — mirrors SECRET_ENV_VARS. */
+  secretEnvVars: Set<string>;
+}
+
+// ─── Service ──────────────────────────────────────────────────────────────────
+
+export class SlackProvisioningService {
+  constructor(private readonly deps: SlackProvisioningServiceDeps) {}
+
+  private redirectUri(): string {
```
Comment (dmcaulay) [inline admin/src/slack-provisioning-service.ts]: `redirectUri()` is hardcoded to `${appBaseUrl}/admin/provision/complete` for every caller, including the new per-agent `connect-slack` flow. Compare with `GithubProvisioningService`, where the route handler supplies a per-agent `redirectUri`/`setupUrl` (see `admin-ui.ts:2321-2322`, which correctly builds `${appBaseUrl}/admin/agents/${agentId}/connect-github/callback`). Because this method isn't parameterized, the Slack app manifest created by `POST /admin/agents/:id/connect-slack` always points its OAuth redirect at the legacy `/admin/provision/complete` route (`admin-ui.ts:2789`, `isAdmin`-gated) rather than the new `GET /admin/agents/:id/connect-slack/callback` route added in this PR — so that new route is unreachable by real Slack OAuth traffic. The smoke tests at `admin-ui.smoke.test.ts:4666+` call the route directly via `app.request()`, which doesn't catch this since it bypasses the actual OAuth redirect. Suggest threading a per-agent `redirectUri` through the same way the GitHub service does.

**Review by dmcaulay — COMMENTED** [review]
Verdict: COMMENT — the prior docs/agent.md finding is fixed (github-provisioning-service.ts's method names now match). New finding: SlackProvisioningService.redirectUri() is hardcoded to the legacy /admin/provision/complete URL for every caller, unlike GithubProvisioningService, which takes a per-agent redirectUri from its caller. That means the new POST /admin/agents/:id/connect-slack flow's Slack app manifest always redirects OAuth completion to the legacy admin-only route, not the new GET /admin/agents/:id/connect-slack/callback route this PR adds — so the new callback route is unreachable via real Slack traffic in production.

File: docs/agent.md [hunk 3]
```diff
@@ -286,6 +286,8 @@ The constant `BAKED_MARKETPLACES_ROOT` and function `discoverBakedMarketplaces()
...
.... Exports `PROVISION_STATE_COOKIE` and `PROVISION_STATE_TTL_SECONDS` constants (shared with admin-ui.ts routes); three core methods: `createAppManifest(agentName, xoxpToken)` — low-level builder called by both provisioning paths, returns raw Slack app credentials; `signProvisionState(agentId, creds, extra?)` — JWT signer for the provision-state cookie, exposes the extra field so the wizard can fold in its `githubOrg` value; `startConnect(agentId, xoxpToken)` — mirrors the wizard's Slack path, validates agent existence, calls `createAppManifest()`, and returns a signed token + OAuth URL; `completeConnect(rawStateCookie, code)` — verifies the provision-state token, exchanges the OAuth code for SLACK_BOT_TOKEN, and returns outcome: `invalid_state` (clear cookie), `missing_code` (preserve cookie for retry), `exchange_failed` (clear cookie), `reinstalled` (SLACK_APP_TOKEN already set; redirect to agent page), or `needs_app_token` (proceed to xapp-token entry); `saveAppToken(agentId, xappToken)` — stores SLACK_APP_TOKEN and reconciles system crons, same as the wizard's final step. No HTTP mechanics (reading/writing cookies, redirects, HTML rendering) — those stay in the route handlers. |
```
Comment (dmcaulay) [inline docs/agent.md]: Stale signatures — `createAppManifest`, `startConnect`, and `completeConnect` all now take a third `redirectUri`/`fallbackRedirectUri` parameter (see `admin/src/slack-provisioning-service.ts` lines 134, 194, 257), which is exactly the fix from the prior review round (the unreachable-callback bug). This row wasn't updated to reflect it — e.g. `startConnect(agentId, xoxpToken)` should read `startConnect(agentId, xoxpToken, redirectUri)`.

**Review by dmcaulay — COMMENTED** [review]
Verdict: COMMENT — both prior findings (docs/agent.md GitHub method names, connect-slack redirect URI unreachable-callback bug) are now fixed and verified. One new, low-severity finding: docs/agent.md's SlackProvisioningService row still shows the pre-fix 2-argument signatures instead of the new 3-argument (redirectUri) ones. Extraction, routing, auth scoping, and tests remain solid.

**Review by dmcaulay — APPROVED** [review]
Verdict: APPROVE — all three prior findings across three review rounds (docs/agent.md GitHub method names, connect-slack redirect URI unreachable-callback bug, stale Slack doc signatures) are now fixed and verified. Extraction is behavior-preserving, auth scoping and redirect-URI wiring on all six new routes are correct, and test isolation is clean.
